### PR TITLE
Openshift APM Server workaround

### PR DESCRIPTION
**Pull request by barkbay**
Fixes for #1410

### CODE REVIEW

File: docs/openshift.asciidoc [hunk 1]
```diff
@@ -3,18 +3,26 @@
...
 This page shows how to run ECK on OpenShift.
 
-NOTE: Only Elasticsearch and Kibana are supported on OpenShift.
+* <<{p}-openshift-before-you-begin,Before you begin>>
+* <<{p}-openshift-deploy-the-operator,Deploy the operator>>
+* <<{p}-openshift-deploy-elasticsearch,Deploy an Elasticsearch instance with a route>>
+* <<{p}-openshift-deploy-kibana,Deploy a Kibana instance with a route>>
+* <<{p}-openshift-apm,Deploy an APM Server instance with a route>>
+
+NOTE: Only Elasticsearch and Kibana are compatible with the `restricted` https://docs.openshift.com/container-platform/4.1/authentication/managing-security-context-constraints.html[Security Context Constraint]. To run the APM Server on OpenShift you must allow the POD to run with the `anyuid` SCC as it is described in <<{p}-openshift-apm,Deploy an APM Server instance with a route>>
```
Comment (alaudazzi) [inline docs/openshift.asciidoc]: as it is described => as described

File: docs/openshift.asciidoc [hunk 2]
```diff
@@ -142,3 +152,75 @@ Use the following command to get the hosts of each `Route`:
 ----
 oc get route -n elastic
 ----
+
+[float]
+[id="{p}-openshift-apm"]
+=== Deploy an APM Server instance with a route
+
+It is currently not possible to run the APM Server with the `restricted` SCC. But as a workaround you may allow the Pod to run with the default `uid 1000` by assigning it to the `anyuid` SCC:
```
Comment (alaudazzi) [inline docs/openshift.asciidoc]: A possible workaround is to allow the Pod to run with the default `uid 1000`

File: docs/openshift.asciidoc [hunk 3]
```diff
@@ -142,3 +152,75 @@ Use the following command to get the hosts of each `Route`:
...
+    termination: passthrough # the APM Server is the TLS endpoint
+    insecureEdgeTerminationPolicy: Redirect
+  to:
+    kind: Service
+    name: apm-server-sample-apm-http
+EOF
+----
++
+You can check that the Pod of the APM Server is using the correct SCC with the following command:
++
```
Comment (alaudazzi) [inline docs/openshift.asciidoc]: To check that the Pod of the APM Server is using the correct SCC, use the following command:

File: docs/openshift.asciidoc [hunk 4]
```diff
@@ -3,18 +3,26 @@
...
 This page shows how to run ECK on OpenShift.
 
-NOTE: Only Elasticsearch and Kibana are supported on OpenShift.
+* <<{p}-openshift-before-you-begin,Before you begin>>
+* <<{p}-openshift-deploy-the-operator,Deploy the operator>>
+* <<{p}-openshift-deploy-elasticsearch,Deploy an Elasticsearch instance with a route>>
+* <<{p}-openshift-deploy-kibana,Deploy a Kibana instance with a route>>
+* <<{p}-openshift-apm,Deploy an APM Server instance with a route>>
+
+NOTE: Only Elasticsearch and Kibana are compatible with the `restricted` https://docs.openshift.com/container-platform/4.1/authentication/managing-security-context-constraints.html[Security Context Constraint]. To run the APM Server on OpenShift you must allow the POD to run with the `anyuid` SCC as it is described in <<{p}-openshift-apm,Deploy an APM Server instance with a route>>
```
Comment (anyasabo) [inline docs/openshift.asciidoc]:
```suggestion
NOTE: Only Elasticsearch and Kibana are compatible with the `restricted` https://docs.openshift.com/container-platform/4.1/authentication/managing-security-context-constraints.html[Security Context Constraint]. To run the APM Server on OpenShift you must allow the Pod to run with the `anyuid` SCC as described in <<{p}-openshift-apm,Deploy an APM Server instance with a route>>
```

**Comment by barkbay** [comment]
Jenkins test this please

**Pull request merged**